modules/currency-l0/src/main/scala/org/tessellation/currency/l0/StateChannel.scala

```diff
@@ -31,7 +32,9 @@ object StateChannel {
         .awakeEvery[F](awakePeriod)
         .evalMap(_ => services.globalL0.pullGlobalSnapshots)
         .evalMap {
-          case Left((snapshot, state)) => storages.lastGlobalSnapshot.setInitial(snapshot, state)
+          case Left((snapshot, state)) =>
+            storages.lastGlobalSnapshot.setInitial(snapshot, state) >> services.sentStateChannelBinaryTrackingService
+              .updateByGlobalSnapshot(snapshot)
```

Comment:
Not sure if you need to add it for this case as it only happens the first time a global snapshot is received.